phoenix-hive/src/main/java/org/apache/phoenix/hive/util/PhoenixUtil.java

```diff
@@ -0,0 +1,163 @@
+import com.google.common.base.Preconditions;
+
+public class PhoenixUtil {
+    static Log LOG = LogFactory.getLog(PhoenixUtil.class.getName());
+
+    public static boolean createTable(Connection conn, String TableName,
+            Map<String, String> fields, String[] pks, boolean addIfNotExists, int salt_buckets,
+            String compression,int versions_num) throws SQLException, MetaException {
+        Preconditions.checkNotNull(conn);
+        if (pks == null || pks.length == 0) {
+            throw new SQLException("Phoenix Table no Rowkeys specified in "
+                    + HiveConfigurationUtil.PHOENIX_ROWKEYS);
+        }
+        for (String pk : pks) {
+            String val = fields.get(pk.toLowerCase());
+            if (val == null) {
+                throw new MetaException("Phoenix Table rowkey " + pk
+                        + " does not belong to listed fields ");
+            }
+            val += " not null";
+            fields.put(pk, val);
+        }
+
+        StringBuffer query = new StringBuffer("CREATE TABLE ");
+        if (addIfNotExists) {
+            query.append("IF NOT EXISTS ");
+        }
+        query.append(TableName + " ( ");
+        Joiner.MapJoiner mapJoiner = Joiner.on(',').withKeyValueSeparator(" ");
+        query.append(" " + mapJoiner.join(fields));
+        if (pks != null && pks.length > 0) {
+            query.append(" CONSTRAINT pk PRIMARY KEY (");
+            Joiner joiner = Joiner.on(" , ");
+            query.append(" " + joiner.join(pks) + " )");
+        }
+        query.append(" )");
+        if (salt_buckets > 0) {
+            query.append(" SALT_BUCKETS = " + salt_buckets);
+        }
+        if (compression != null) {
+            query.append(" ,COMPRESSION='GZ'");
+        }
+        if (versions_num > 0) {
+            query.append(" ,VERSIONS="+versions_num);
+        }
+        System.out.println("CREATED QUERY " +query.toString());
+        LOG.info("Create table query statement " + query.toString());
+        return createTable(conn, query.toString());
+    }
+
+    public static boolean createTable(Connection conn, String query) throws SQLException {
+        Preconditions.checkNotNull(conn);
+        return conn.createStatement().execute(query);
+    }
+
+    public static boolean findTable(Connection conn, String name) throws SQLException {
+        Preconditions.checkNotNull(conn);
+        Preconditions.checkNotNull(name);
+        DatabaseMetaData dbm = conn.getMetaData();
+        ResultSet rs = dbm.getTables(null, null, name, null);
+        LOG.info("looking for table");
+        if (rs.next()) {
+            LOG.info("found the table " + rs.getString("TABLE_NAME"));
+            while (rs.next()) {
+                LOG.info("found the table " + rs.getString("TABLE_NAME"));
+            }
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean testTable(Connection conn, String name, Map<String, String> fields)
```

Comment:
Move out any test-specific methods to test util classes instead of having them in runtime classes.